snap/squashfs/squashfs.go

```diff
@@ -110,6 +113,11 @@ func (s *Snap) Install(targetPath, mountDir string, opts *snap.InstallOptions) (
 		}
 	}
 
+	if opts.SkipOnDiskSnapValidation && osutil.FileExists(targetPath) {
```

Comment:
we should have a comment here about why/when this is used, there are quite a few special cases in this function already and if we are adding more we should make it clear, though tbh now looking back this function's structure is not very readable as it has the form:

```go
    // if ABC is true do foo()
    if !ABC {
        bar()
    }

    foo()
```

which to me is confusing to follow due to the negatives and having the comment describing foo() not above foo() but above bar() for example. Though that's not your fault, but we should at least not make this function more confusing I think by not including any comments about this

Reply:
I'm not sure I see that structure in this function. Do you mean that `bar()` is calling `os.Symlink()` and `foo()` is calling `osutil.CopyFile()`?

Reply:
See 
```
we are operating on a overlayfs based system (e.g. live
	// installer) use symbolic links.
```

the if condition `overlayRoot == ""` is true when we _are not_ on an overlayfs system, and we use symbolic links further below